sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -289,14 +289,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
   }
 
   def namedArgumentsNotEnabledError(functionName: String, argumentName: String): Throwable = {
-    new AnalysisException(
+    new SparkUnsupportedOperationException(
```

Comment:
Why are we changing this from an `AnalysisException` to a `SparkUnsupportedOperationException`?

Another way to ask kind of the same thing: why are we not using `SparkUnsupportedOperationException` in OSS?

Reply:
It slipped in, I was thinking whether my exception should be one or the other, so I accidentally changed this.